### ISSUE TITLE
🌟

### ISSUE THREAD

**Issue by GBHU753** [HiEventsDev/Hi.Events#206]
**Is your feature request related to a problem? Please describe.**
I've had good look around and I can't seem to find a way to get all the event ids that are public

**Describe the solution you'd like**
The `{{host}}/events` endpoint moved to being public.

**Describe alternatives you've considered**
I couldn't come up with anything else.

**Additional context**
N/A

**Comment by GBHU753** [comment]
Maybe I could make an account with the specific purpose of accessing all the events

**Comment by daveearley** [comment]
Hi @GBHU753,

This is [currently in progress](https://github.com/orgs/HiEventsDev/projects/1?pane=issue&itemId=73943499) as part of adding public organizer pages. It should be complete in the next couple of weeks.

**Comment by grablair** [comment]
@daveearley did this get put on the backburner? Did you make any progress on it? If not, I may pick it up.

**Comment by daveearley** [comment]
Hey @grablair,

I had to pause work to focus on the [migration from tickets to products](https://github.com/HiEventsDev/Hi.Events/pull/239). I made progress on preparing for public organizer pages, but I haven't implemented the public events endpoint yet.

Feel free to pick it up, and let me know if you have any questions about the implementation!

Thanks!

**Comment by oneplacepress** [comment]
> I had to pause work to focus on the [migration from tickets to products](https://github.com/HiEventsDev/Hi.Events/pull/239). I made progress on preparing for public organizer pages, but I haven't implemented the public events endpoint yet.

Hi, has this feature been abandoned?  I'm trying to understand how a user can see a list/calendar of upcoming public events without logging in.  Why would anybody register on my site if they can't see a list of events available?  Thanks.

**Comment by oneplacepress** [comment]
> [@daveearley](https://github.com/daveearley) did this get put on the backburner? Did you make any progress on it? If not, I may pick it up.

Ever work on this?  Any idea how to share a list of public events with somebody?  I don't understand how an events management platform doesn't have a calendar list of upcoming events for non-authenticated users to browse before deciding to sign up.

**Comment by daveearley** [comment]
Hi @oneplacepress,

This work hasn’t started yet. It’s not a frequently requested feature, so it hasn’t been a priority up to now. 

That said, we do plan to add a public API endpoint for fetching events, as it can be useful for custom calendars or embedding events on organizer websites. There is an undocumented endpoint for fetching a single organizers event. You can find that [here](https://github.com/HiEventsDev/Hi.Events/pull/362/files#diff-11f02f973a1aa5eaa5b4221c3e10efdce1a4e9665ab222d792c858a6fed7ccb2R274).

**Comment by cgonIT** [comment]
> Hi [@oneplacepress](https://github.com/oneplacepress),
> 
> This work hasn’t started yet. It’s not a frequently requested feature, so it hasn’t been a priority up to now.
> 
> That said, we do plan to add a public API endpoint for fetching events, as it can be useful for custom calendars or embedding events on organizer websites. There is an undocumented endpoint for fetching a single organizers event. You can find that [here](https://github.com/HiEventsDev/Hi.Events/pull/362/files#diff-11f02f973a1aa5eaa5b4221c3e10efdce1a4e9665ab222d792c858a6fed7ccb2R274).

Hi @daveearley , any update on this? Would be very usefull to have this feature so everyone can see past/next events without the need of have an per-event web link, just enter in the "events.example.com" to see the list of events.

Thanks.

**Comment by daveearley** [comment]
> Would be very usefull to have this feature so everyone can see past/next events without the need of have an per-event web  link, just enter in the "events.example.com" to see the list of events.

There is a [big change](https://github.com/HiEventsDev/Hi.Events/pull/615) coming soon that will allow you to create a public event listing page for your organizer.

**Comment by cgonIT** [comment]
> > Would be very usefull to have this feature so everyone can see past/next events without the need of have an per-event web  link, just enter in the "events.example.com" to see the list of events.
> 
> There is a [big change](https://github.com/HiEventsDev/Hi.Events/pull/615) coming soon that will allow you to create a public event listing page for your organizer.

Hi @daveearley any update on this? Asking just to know because I'm not very familiar with github way of working, and I see the "Feature: Add public organinzer pages #615" and lot of changes, but I don't know if they are published. 
Once this "big change" will be published, I only need to re-deploy my docker container to get the las updates, right?
Thank you so much. I don't know if there is any way to help you with something.

**Comment by daveearley** [comment]
@cgonIT yes, it has been released. You will need to pull the latest code or Docker image. You may also need to run `php artisan migrate` in the container depending on your setup.

**Comment by cgonIT** [comment]
> [@cgonIT](https://github.com/cgonIT) yes, it has been released. You will need to pull the latest code or Docker image. You may also need to run `php artisan migrate` in the container depending on your setup.
Thank you once for your great job.


**UPDATING THIS TICKET**
This is the proccedure I did to upgrade my selfhosted server:

_Scenario:_
- Ubuntu server running docker.
- I cloned the git repo and using the "Docker >all-in-one" version.
- Postgress db in another docker container.

_Update steps I did:_
- Backup of the hi.eventes folder where configuration files are.
- docker compose down.
- Delete the docker image "all-in-one".
- Delete the folder hi.events.
- Git clone of  hi.Events repo.
- Replace from backup folder the next items: 
-- "\docker\all-in-one\.envs"
-- "\docker\all-in-one\docker-compose.yml"
- docker compose up -d

Now I've my selfhosted version updated. 
Hope it helps.

**Comment by cgonIT** [comment]
> You may also need to run

Hi @daveearley ! I thougt everything was ok but...After the update, attendees are getting an error in the GUI when they clic in "Place Order":

<img width="1173" height="139" alt="Image" src="https://github.com/user-attachments/assets/d2abd0ae-8f45-4b53-8ac7-772cfadd5330" />

But the booking ins launched as I receive the email with the request.

My Deployment Scenario detailled:

<img width="1124" height="117" alt="Image" src="https://github.com/user-attachments/assets/73e70815-41ec-48b3-ac94-805374d29015" />
- 2 Docker containers running in Ubuntu Server.
- 1 is the Hi.Events container
- 1 is the PostgreSql container
Both deployed from the docker > all-in-one method.

If I do:
"docker exec -it postgres_container_id bash" I got connected to the db container but unable to run "php artisan migrate".

<img width="418" height="61" alt="Image" src="https://github.com/user-attachments/assets/86fbb3ce-41b1-469e-b965-285ce45a73df" />

In the hi.events container, if I move to "backend" folder and run the same command:

<img width="639" height="227" alt="Image" src="https://github.com/user-attachments/assets/3bade8e3-64a0-405c-9e71-9a98d8cbc9f3" />

Nothing to migrate.

So I decide to create a new event since there are things from your code that has been changed, just to see if the problem is in or not, but even with a new event the "Server Error" message appears. I receive the email in the admin mailbox as a new booking has been done, but the Server Error message is worrying me.

So don't know if the problem is related or not.

From Docker Portainer I can see live logs from server and I see this lines:

<img width="1373" height="191" alt="Image" src="https://github.com/user-attachments/assets/b2e91ef9-9941-4b54-8cb9-d432f783b048" />

And in the Microsoft Edge Broswer from where the book is done I see this (A 500 status code error basicaly):

<img width="1475" height="968" alt="Image" src="https://github.com/user-attachments/assets/2d9c4a7e-9809-4d11-98e4-53b964efbf77" />

My apologies if I miss something due I'm not a fully expert on this.

Do you know what am I missing?

Thank you!
